src/rpcrequest-payload.ts

```diff
@@ -12,13 +15,14 @@ import { Collation } from './collation';
 
 const STATUS = {
   BY_REF_VALUE: 0x01,
-  DEFAULT_VALUE: 0x02
+  DEFAULT_VALUE: 0x02,
+  ENCRYPTED_VALUE: 0x08,
 };
 
 /*
   s2.2.6.5
  */
-class RpcRequestPayload implements Iterable<Buffer> {
+class RpcRequestPayload implements AsyncIterable<Buffer> {
```

Comment:
@MichaelSun90 This change here (switching `RPCRequestPayload` from a `Iterable` to a `AsyncIterable`) should be extracted into a separate PR.

The change will allow us to e.g. have `AsyncIterable` objects as values for parameters to `nvarchar(max)`, `varchar(max)`, `varbinary(max)`, and TVPs rows.